### PR TITLE
E621 Command

### LINKED ISSUE #78: Add an e621 command

### PR DESCRIPTION
**Pull request by Doges**
haha @pedrofracassi i did it

Closes #78

### CODE REVIEW

**Comment by bolsomito** [comment]
why is furry a label smh

**Comment by pedrofracassi** [comment]
why the fuck do we have a furry label?

**Comment by pedrofracassi** [comment]
Closes #78

**Comment by Doges** [comment]
@pedrofracassi why tf you don't test this shit and approve it?

**Comment by pedrofracassi** [comment]
@Doges cause I'm working on a school project.

File: src/commands/furry/e621.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            message.channel.send(embed)
+          }
+        })
+    } else {
+      embed.setColor(Constants.ERROR_COLOR)
+      embed.setTitle('You need to use this command in a NSFW channel')
+      embed.setDescription(`**Usage:** ${process.env.PREFIX}${this.name}`)
+      message.channel.send(embed)
+    }
+    message.channel.stopTyping()
```
Comment (bolsomito) [inline src/commands/furry/e621.js]: `message.channel.send(embed).then(() => message.channel.stopTyping())` to remove duped code (lines 21 and 28)

Comment (davipatury) [reply src/commands/furry/e621.js]: and use await

File: src/commands/furry/e621.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    message.channel.startTyping()
+    const embed = new SwitchbladeEmbed(message.author)
+    if (message.channel.nsfw) {
+      booru.search('e621.net', ['rating:e'], {limit: 1, random: true})
+        .then(booru.commonfy)
+        .then(images => {
+          for (let image of images) {
+            embed.setImage(image.common.file_url)
+            embed.setDescription('Here\'s your yiff!')
+            message.channel.send(embed).then(message => message.channel.stopTyping())
```
Comment (bolsomito) [inline src/commands/furry/e621.js]: remove this line

File: src/commands/furry/e621.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+          }
+        })
+    } else {
+      embed.setColor(Constants.ERROR_COLOR)
+      embed.setTitle('You need to use this command in a NSFW channel')
+      embed.setDescription(`**Usage:** ${process.env.PREFIX}${this.name}`)
+      message.channel.send(embed).then(message => message.channel.stopTyping())
```
Comment (bolsomito) [inline src/commands/furry/e621.js]: remove this line

Comment (Doges) [reply src/commands/furry/e621.js]: WHICH LINE

File: src/commands/furry/e621.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    }
```
Comment (bolsomito) [inline src/commands/furry/e621.js]: add `message.channel.send(embed).then(() => message.channel.stopTyping())`

Comment (Doges) [reply src/commands/furry/e621.js]: but wait, if i remove the line, the embed don't gonna send in normal message 🤔

File: src/commands/furry/e621.js (hunk 3 above)
Comment (bolsomito) [reply src/commands/furry/e621.js]: 28

File: src/commands/furry/e621.js (hunk 2 above)
Comment (bolsomito) [reply src/commands/furry/e621.js]: line 21

File: src/commands/furry/e621.js (hunk 4 above)
Comment (bolsomito) [reply src/commands/furry/e621.js]: insert at the end of the if else case

Comment (davipatury) [reply src/commands/furry/e621.js]: user async/await

Comment (Doges) [reply src/commands/furry/e621.js]: not needed, already solved 🤔

File: src/commands/furry/e621.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    message.channel.startTyping()
+    const embed = new SwitchbladeEmbed(message.author)
+    if (message.channel.nsfw) {
+      booru.search('e621.net', ['rating:e'], {limit: 1, random: true})
+        .then(booru.commonfy)
+        .then(images => {
+          for (let image of images) {
+            embed.setImage(image.common.file_url)
+            embed.setDescription('Here\'s your yiff!')
+          }
```
Comment (davipatury) [inline src/commands/furry/e621.js]: why the fuck you're looping this?

Comment (Doges) [reply src/commands/furry/e621.js]: not a loop totally, if theres an image, it gets only 1, than it sends, but tbh, idk why i put that.

Comment (Doges) [reply src/commands/furry/e621.js]: also, w/o this the command doesn't works

File: src/commands/furry/e621.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+  async run (message) {
+    message.channel.startTyping()
+    const embed = new SwitchbladeEmbed(message.author)
+    if (message.channel.nsfw) {
+      const images = await booru.search('e621.net', ['rating:e'], {limit: 1, random: true})
+        .then(booru.commonfy)
+      for (let image of images) {
+        embed.setImage(image.common.file_url)
+        embed.setDescription('Here\'s your yiff!')
```
Comment (davipatury) [inline src/commands/furry/e621.js]:
```js
const [ image ] = await booru.search('e621.net', ['rating:e'], {limit: 1, random: true}).then(booru.commonfy)
embed.setImage(image.common.file_url)
embed.setDescription('Here\'s your yiff!')```

Comment (Doges) [reply src/commands/furry/e621.js]: it doesn't works

Comment (pedrofracassi) [reply src/commands/furry/e621.js]: Same for your english

Comment (Doges) [reply src/commands/furry/e621.js]: define that

**Comment by Doges** [comment]
fuck

**Comment by bolsomito** [comment]
ooooo snap

File: src/commands/furry/e621.js (hunk 6 above)
Comment (Doges) [reply src/commands/furry/e621.js]: ignore, the embeds was glitched

**Comment by Doges** [comment]
remaking the command

**Pull request closed without merging**